bot/exts/utilities/colour.py

```diff
@@ -0,0 +1,257 @@
+                message=f"RGB values can only be from 0 to 255. User input was: `{red, green, blue}`."
+            )
+        rgb_tuple = (red, green, blue)
+        await self.send_colour_response(ctx, rgb_tuple)
+
+    @colour.command()
+    async def hsv(self, ctx: commands.Context, hue: int, saturation: int, value: int) -> None:
+        """Create an embed from an HSV input."""
+        if (hue not in range(361)) or any(c not in range(101) for c in (saturation, value)):
+            raise commands.BadArgument(
+                message="Hue can only be from 0 to 360. Saturation and Value can only be from 0 to 100. "
+                f"User input was: `{hue, saturation, value}`."
+            )
+        hsv_tuple = ImageColor.getrgb(f"hsv({hue}, {saturation}%, {value}%)")
+        await self.send_colour_response(ctx, hsv_tuple)
+
+    @colour.command()
+    async def hsl(self, ctx: commands.Context, hue: int, saturation: int, lightness: int) -> None:
+        """Create an embed from an HSL input."""
+        if (hue not in range(361)) or any(c not in range(101) for c in (saturation, lightness)):
+            raise commands.BadArgument(
+                message="Hue can only be from 0 to 360. Saturation and Lightness can only be from 0 to 100. "
+                f"User input was: `{hue, saturation, lightness}`."
+            )
+        hsl_tuple = ImageColor.getrgb(f"hsl({hue}, {saturation}%, {lightness}%)")
+        await self.send_colour_response(ctx, hsl_tuple)
+
+    @colour.command()
+    async def cmyk(self, ctx: commands.Context, cyan: int, magenta: int, yellow: int, key: int) -> None:
+        """Create an embed from a CMYK input."""
+        if any(c not in range(101) for c in (cyan, magenta, yellow, key)):
+            raise commands.BadArgument(
+                message=f"CMYK values can only be from 0 to 100. User input was: `{cyan, magenta, yellow, key}`."
+            )
+        r = round(255 * (1 - (cyan / 100)) * (1 - (key / 100)))
+        g = round(255 * (1 - (magenta / 100)) * (1 - (key / 100)))
+        b = round(255 * (1 - (yellow / 100)) * (1 - (key / 100)))
+        await self.send_colour_response(ctx, (r, g, b))
+
+    @colour.command()
+    async def hex(self, ctx: commands.Context, hex_code: str) -> None:
+        """Create an embed from a HEX input."""
+        if hex_code[0] != "#":
+            hex_code = f"#{hex_code}"
+
+        if len(hex_code) not in (4, 5, 7, 9) or any(digit not in string.hexdigits for digit in hex_code[1:]):
+            raise commands.BadArgument(
+                message=f"Cannot convert `{hex_code}` to a recognizable Hex format. "
+                "Hex values must be hexadecimal and take the form *#RRGGBB* or *#RGB*."
+            )
+
+        hex_tuple = ImageColor.getrgb(hex_code)
+        if len(hex_tuple) == 4:
+            hex_tuple = hex_tuple[:-1]  # color must be RGB. If RGBA, we remove the alpha value
+        await self.send_colour_response(ctx, hex_tuple)
```

Comment:
Bit of a nitpick too since this is probably unlikely to happen and not that easy to notice, but #RRGGBB and #RRGGBBAA can produce [different colors](https://codepen.io/hedyhli/pen/XWaBZzd):

![2A2217CC-6DD3-4243-BFD7-690390803BE4](https://user-images.githubusercontent.com/50042066/141235615-7bd514da-d108-410e-8940-148d913e9a2d.jpeg)

![74DFC550-B724-4473-A121-ADC6BAF3EB65](https://user-images.githubusercontent.com/50042066/141235146-865375bd-3dda-4fd6-b243-a58bdd9936ae.jpeg)

It seems the alpha value is ignored, but it still shows up in "Hex information for [**here**]". I'm thinking we should strip out the alpha value in `ctx.args` (because it's used by `send_colour_response`) so it eliminates confusion as to two different values producing the same output embed.

That said, feel free to support the alpha value in hex mode though (maybe with RGBA) if you want and have time 😄.

Reply:
😅 I'm going to leave the alpha values for someone else.....but I did strip them off of the embed. Should be fixed with: 1914905